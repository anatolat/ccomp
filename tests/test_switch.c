
void test_simple(int i) {
 switch (i) {
   case 1: 
   case 3: 
          printf("1 or 3\n"); 
          break;

   case 4: 
   case 5: 
          printf("4 or 5\n"); 
          break;

  default:
          printf("x %d\n", i);
          break;
  }
}

void test_default(int i) {
 switch (i) {
  default:
          printf("x %d\n", i);
  }
}
 

int main() {
 for (int i = 0; i < 10; ++i) {
	test_simple(i);
 }
 printf("\n");
 for (int i = 0; i < 10; ++i) {
	test_default(i);
 }
 printf("\n");
}