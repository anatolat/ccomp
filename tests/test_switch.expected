x 0
1 or 3
x 2
1 or 3
4 or 5
4 or 5
x 6
x 7
x 8
x 9

x 0
x 1
x 2
x 3
x 4
x 5
x 6
x 7
x 8
x 9

